### PR TITLE
feat: Have travis bump submodule commits in docker build repository

### PR DESCRIPTION
**Pull request by Bouncey**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,5 +21,11 @@ before_script:
   - git config --global user.email team+camperbot@freeCodeCamp.org
   - git config --global user.name "CamperBot"
 
-script:
-  - npm run test
+jobs:
+  include:
+    - stage: Test the things
```
Comment (raisedadead) [inline .travis.yml]:
```suggestion
    - stage: Unit and Integration tests
```

File: .travis.yml [hunk 2]
```diff
@@ -21,5 +21,11 @@ before_script:
...
+      script: npm test
+
+    - stage: Trigger build
```
Comment (raisedadead) [inline .travis.yml]:
```suggestion
    - stage: Artifacts builds
```

File: .travis.yml [hunk 3]
```diff
@@ -21,5 +21,11 @@ before_script:
...
+      if: branch = master && type != pull_request
```
Comment (raisedadead) [inline .travis.yml]: We should have a issue tracking "CI Stage for production"

**Review by raisedadead — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 4]
```diff
@@ -21,5 +21,11 @@ before_script:
...
+      script: bash trigger-build.sh
```
Comment (raisedadead) [inline .travis.yml]:
```suggestion
      script: bash trigger-build.sh
 
```

Newline. UNIX doesn't play well at times.

File: trigger-build.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+if [ -n "$DOCKER_PUSH_API_TOKEN" ]
+then
+  git clone https://github.com/"$DOCKER_PUSH_REPO".git docker-push-repo
+
+  cd docker-push-repo/freecodecamp
+  git pull --rebase origin master
+  cd ../
+  git add freecodecamp
+  git -c user.name="$DOCKER_PUSH_USER" -c user.email='travis' \
+    commit -m 'Travis CI - bump submodule'
```
Comment (raisedadead) [inline trigger-build.sh]:
```suggestion
    commit -m 'chore(ci): Travis CI - bump submodule'
```

File: trigger-build.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  git push https://"$DOCKER_PUSH_USER":"$DOCKER_PUSH_API_TOKEN"@github.com/"$DOCKER_PUSH_REPO".git master
+fi
+popd
```
Comment (raisedadead) [inline trigger-build.sh]:
```suggestion
popd

```

Newline.

**Review by raisedadead — CHANGES_REQUESTED** [review]

File: trigger-build.sh (hunk 5 above)
Comment (raisedadead) [reply trigger-build.sh]: This one is a nit pick

File: .travis.yml (hunk 3 above)
Comment (Bouncey) [reply .travis.yml]: In my head, I have something that builds the api image, runs the image and queries the api to ensure the correct responses are given.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**